### DISCUSSION TITLE
Fail to install on ubuntu server

### DISCUSSION THREAD

**Discussion by raxp122** [matrix-org/dendrite#2295] [Help & Support]
Hi everyone i recently opened a new ubuntu server and started following this tutorial
https://pcarion.com/posts/matrix-dendrite-install/

After installing go and building dendrite i get this error

```
# runtime/internal/sys
/usr/local/go/src/runtime/internal/sys/stubs.go:9:7: PtrSize redeclared in this block
        /usr/local/go/src/runtime/internal/sys/arch.go:24:38: previous declaration
/usr/local/go/src/runtime/internal/sys/stubs.go:10:24: undefined: Uintreg
/usr/local/go/src/runtime/internal/sys/stubs.go:16:7: StackGuardMultiplier redeclared in this block
        /usr/local/go/src/runtime/internal/sys/arch.go:27:74: previous declaration
```

And so the next step
` ./bin/generate-keys --private-key matrix_key.pem `
tells me that the bin folder is empty, what am i missing, sorry i'm new to this...

**Comment by neilalexander** [comment]
How did you install Go, and which version did you install? Those look like very suspicious error messages not related to Dendrite.

**Reply by spaetz** [reply]
that howto states go 1.15 is needed, while by now go 1.17 is needed, so you'd need to install at least that version on your ubuntu server. You are not saying which Ubuntu version you are on. But you need to be at least on "impish" to have golang 1.17 available as you can see here:
https://packages.ubuntu.com/search?suite=impish&searchon=names&keywords=golang
